guides/source/active_record_migrations.md

```diff
@@ -1307,3 +1623,118 @@ special comment like this:
 ```
 
  [Engines]: engines.html
+
+## Miscellaneous
+
+### Using UUIDs instead of IDs for Primary Keys
+
+By default, Rails uses auto-incrementing integers (often referred to as "IDs")
+as primary keys for database records. However, there are scenarios where using
+Universally Unique Identifiers (UUIDs) as primary keys can be advantageous,
+especially in distributed systems or when integration with external services is
+necessary. UUIDs provide a globally unique identifier without relying on a
+centralized authority for generating IDs.
+
+#### Enabling UUIDs in Rails
+
+Before using UUIDs in your Rails application, you'll need to ensure that your
+database supports storing them. Additionally, you may need to configure your
+database adapter to work with UUIDs.
+
+1. Database Configuration
+
+    Ensure that your database supports storing UUIDs. PostgreSQL natively
+    supports UUIDs with its `uuid` data type, while other databases might
+    require additional configuration.
+
+    Enabling `uuid-ossp` Extension: If you're using PostgreSQL, enable the
```

Comment:
I don't think this extension is required to generate uuid values in Postgres in recent versions (I can look this up) or store data using the uuid data type.

If you'd like a longer explanation I can provide that. My purpose in this comment is to avoid guidance to enable an extension that's not necessary.

Reply:
Yes, as of [version 13](https://www.postgresql.org/docs/release/13.0/):

> Add function [gen_random_uuid()](https://www.postgresql.org/docs/13/functions-uuid.html) to generate version-4 UUIDs (Peter Eisentraut)

In order for the rest of the guide to make sense on older versions of Postgres, pgcrypto needs to be enabled for [`gen_random_uuid()`](https://www.postgresql.org/docs/12/pgcrypto.html#id-1.11.7.34.9). The older uuid-ossp extension provides the [`uuid_generate_v4()`](https://www.postgresql.org/docs/12/uuid-ossp.html) function, and the documentation for it even recommends using pgcrypto for UUIDv4.

Reply:
Thanks so much for pointing this out, I've amended it as follows: 

```
1. Database Configuration

    If you are using a version of PostgreSQL prior to 13, you may still need to
    enable the pgcrypto extension to access the `gen_random_uuid()` function. For
    PostgreSQL 13 and later, this is not necessary.

    For PostgreSQL versions prior to 13:

    ```ruby
    class EnablePgcryptoExtension < ActiveRecord::Migration[8.0]
      def change
        enable_extension "pgcrypto"
      end
    end
    ```
```